adding synthetics apm page (#4130)
DataDog/documentation: content/en/synthetics/apm.md

---
title: Synthetics APM
kind: documentation
beta: true
description: APM and Distributed Tracing with Synthetics
further_reading:
- link: "https://www.datadoghq.com/blog/introducing-synthetic-monitoring/"
  tag: "Blog"
  text: "Introducing Datadog Synthetics"
- link: "tracing/"
  tag: "Documentation"
  text: "APM and Distributed Tracing"

---

<div class="alert alert-warning">Synthetics is in beta for the Datadog US Site. To request access, complete the <a href="https://app.datadoghq.com/synthetics/beta">Datadog Synthetics request form</a>.</div>

{{< img src="synthetics/apm/synthetics-apm.gif" alt="APM and Synthetics" responsive="true" style="width:80%;">}}

## Overview

The APM integration with Synthetics allows you to go from a test run that potentially failed to the root cause of the issue by looking at the trace generated by that very test run.

> Having both network related details thanks to your test and backend information thanks to your trace allows you to access a whole new level of details in the way your application is behaving as experienced by your user.

Reviewer: ```suggestion
Having both network related specifics, thanks to your test, and backend information, thanks to your trace, allows you to access a whole new level of details in the way your application is behaving as experienced by your user.
```

Reviewer 2: From GJ: Having both network related specifics, thanks to your test, and backend, infra and log information, thanks to your trace, allows you to access a whole new level of details in the way your application is behaving as experienced by your user.

## Usage

### Prerequisites

* Your service is [traced on the APM side][1].
* Your service uses an HTTP server.
* Your HTTP server is using a library that supports distributed tracing.

Create a check that hits your traced HTTP server, and Datadog automatically links the trace generated by your server to the corresponding check result.

### How are traces linked to checks?

Datadog uses the distributed tracing protocol and sets up the following HTTP headers:

* `x-datadog-trace-id`: Generated from the Synthetics backend. Allows Datadog to link the trace with the test result.

> * `x-datadog-parent-id`: 0

Reviewer 2: I believe it needs to be `x-datadog-parent-id: 0`

> * `x-datadog-origin`: Synthetics to make sure the generated traces don't affect your APM quotas. See below.

Reviewer 2: It needs to be `x-datadog-origin: synthetics`

> * `x-datadog-sampling-priority`: 1, [to make sure that the Agent keeps the trace][2].

Reviewer 2: I believe it needs to be `x-datadog-sampling-priority: 1`

### How are APM quotas affected?

> The `x-datadog-origin` header specifies to the APM backend that the traces were generated by Synthetics. The generated traces consequently do not impact your classical APM quotas.

Reviewer: ```suggestion
The `x-datadog-origin` header specifies to the APM backend that the traces are Synthetics generated. The generated traces consequently do not impact your classical APM quotas.
```
